actor/src/main/scala/org/apache/pekko/io/dns/internal/TcpDnsClient.scala

```diff
@@ -80,7 +80,10 @@ import pekko.util.ByteString
         }
       }
     case Tcp.PeerClosed =>
+      context.unwatch(connection)
       context.become(idle)
+    case Terminated(`connection`) =>
+      throwFailure("TCP connection terminated without closing (register timeout?)", None)
```

Comment:
Did you notice `TcpOutgoingConnection` doing the same thing here?

https://github.com/apache/incubator-pekko/blob/46e60a61fbabce5e3f36a408bfa3d1fb249eef44/actor/src/main/scala/org/apache/pekko/io/TcpOutgoingConnection.scala#L58-L59

It will send `CommandFailed` to `commander(TcpDnsClient)`

Reply:
Not in the case of the register timeout, there it will just do `context.stop`.

Reply:
Ahh, I see. The connection is ready, and there are only the last two Acks left. Things from TcpOutgoingConnection to TcpConnection. 

Thanks for your explanation.